sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -509,6 +509,10 @@ object ScalaReflection extends ScalaReflection {
                 serializerFor(unwrapped, optType, newPath))
           }
 
+        case t if t <:< localTypeOf[Seq[_]] =>
```

Comment:
add some comment to the definedByConstructorParams case to explain it is sensitive to ordering, and why?